license-tool/src/main.rs

```diff
@@ -0,0 +1,378 @@
+                fs::copy(license_url.path(), &path)
+                    .context(format!("Failed to copy file from '{}'", license_url.path()))?;
+            }
+            "http" | "https" => {
+                let content = reqwest::get(license.license_url.clone())
+                    .await
+                    .context(format!(
+                        "Failed to download file from '{}'",
+                        license.license_url.to_string()
+                    ))?
+                    .text()
+                    .await?;
+                let mut dest = File::create(&path).context(format!(
+                    "Failed to create file '{}'",
+                    path.display().to_string()
+                ))?;
+                io::copy(&mut content.as_bytes(), &mut dest).context(format!(
+                    "Failed to copy content to '{}'",
+                    path.display().to_string()
+                ))?;
+            }
+            _ => bail!(
+                "Invalid scheme for '{}', valid options are: ['file://', 'http://', 'https://']",
+                license.license_url
+            ),
+        };
+    }
+    Ok(())
+}
+
+/// Returns a list of paths to the destination files for the licenses
+fn get_paths<P>(package_license: &PackageLicense, dest: P) -> Result<Vec<String>>
+where
+    P: AsRef<Path>,
+{
+    let mut all_paths = Vec::new();
+    for license in &package_license.licenses {
+        all_paths.push(get_path(&license, &dest)?);
+    }
+
+    Ok(all_paths)
+}
+
+/// Returns the path to the destination file for the license
+fn get_path<P>(license: &License, base: P) -> Result<String>
+where
+    P: AsRef<Path>,
+{
+    let base = base.as_ref();
+    let path = match &license.path {
+        // If the path was given, use it
+        Some(path) => base.join(path),
+        // Otherwise, use the original file's name
+        None => base.join(
+            license
+                .license_url
+                .path_segments()
+                .context("Failed to get path segments")?
+                .last()
+                .context("Failed to get last segment")?
+                .to_string(),
+        ),
+    };
+
+    Ok(path.display().to_string())
+}
+
+/// Parses a map of `PackageLicense` objects from an array of bytes
+fn parse_licenses_file<P>(licenses_file: P) -> Result<HashMap<String, PackageLicense>>
+where
+    P: AsRef<Path>,
+{
+    let licenses_file = licenses_file.as_ref();
+    Ok(toml::from_slice(&fs::read(&licenses_file).context(
+        format!("Failed to read file '{}'", licenses_file.display()),
+    )?)?)
+}
+
+/// Replaces the ENV variables when present in a file URL
+fn replace_env_vars<S>(path: S) -> Result<Url>
+where
+    S: AsRef<str>,
+{
+    let mut segments: Vec<String> = vec!["file://".to_string()];
+    let path = path.as_ref().to_string();
+    let regex = Regex::new(ENV_PATTERN)?;
+    let file_path = path.split("://").skip(1).next().expect("Invalid path");
```

Comment:
```suggestion
    let file_path = path.split("://").nth(1).expect("Invalid path");
```
Another clippy suggestion.